File: api/lookup.py
from http.server import BaseHTTPRequestHandler
import requests
import json
import urllib.parse

class handler(BaseHTTPRequestHandler):
    def do_GET(self):
        try:
            # Parse query parameters
            parsed = urllib.parse.urlparse(self.path)
            params = urllib.parse.parse_qs(parsed.query)
            
            # Get mobile number from query
            mobile = params.get('mobile', [None])[0]
            
            if not mobile:
                self.send_response(400)
                self.send_header('Content-type', 'application/json')
                self.end_headers()
                self.wfile.write(json.dumps({
                    "error": "Mobile number is required",
                    "usage": "/api/lookup?mobile=9811790070"
                }).encode())
                return
            
            # Call the original API
            original_url = f"https://ethicaltabbo.in/api/lookup?key=Sahil&mobile={mobile}"
            response = requests.get(original_url, timeout=15)
            
            if response.status_code == 200:
                data = response.json()
                
                # Clean the data - remove unwanted fields
                if 'telegram' in data:
                    del data['telegram']
                if 'channel' in data:
                    del data['channel']
                if 'credit' in data:
                    del data['credit']
                if 'api_info' in data:
                    del data['api_info']
                
                # Remove unwanted fields from each record
                if 'data' in data and data['data']:
                    for record in data['data']:
                        if 'id' in record:
                            del record['id']
                        if 'alt_number' in record:
                            del record['alt_number']
                
                # Add credit to response
                data['credit'] = "GENIUS HACKER 29 API"
                data['developer'] = "ADIBHAI"
                data['youtube'] = "https://youtube.com/@geniushacker29"
                
                self.send_response(200)
                self.send_header('Content-type', 'application/json')
                self.end_headers()
                self.wfile.write(json.dumps(data, indent=2, ensure_ascii=False).encode())
            else:
                self.send_response(response.status_code)
                self.send_header('Content-type', 'application/json')
                self.end_headers()
                self.wfile.write(json.dumps({
                    "error": "Original API returned error",
                    "status": response.status_code
                }).encode())
                
        except Exception as e:
            self.send_response(500)
            self.send_header('Content-type', 'application/json')
            self.end_headers()
            self.wfile.write(json.dumps({
                "error": str(e)
            }).encode())

File: api/lookup.js
// GENIUS HACKER 29 API - Node.js Version

export default async function handler(req, res) {
    // Enable CORS
    res.setHeader('Access-Control-Allow-Origin', '*');
    res.setHeader('Access-Control-Allow-Methods', 'GET, OPTIONS');
    res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

    // Handle preflight
    if (req.method === 'OPTIONS') {
        return res.status(200).end();
    }

    try {
        // Get mobile number from query
        const { mobile } = req.query;

        // If no mobile parameter
        if (!mobile) {
            return res.status(400).json({
                status: false,
                error: 'Mobile number required',
                usage: '/api/lookup?mobile=9876543210'
            });
        }

        // Validate mobile number
        if (!/^\d{10}$/.test(mobile)) {
            return res.status(400).json({
                status: false,
                error: 'Invalid mobile number',
                message: 'Please send exactly 10 digits'
            });
        }

        // Call the original API
        const originalUrl = `https://ethicaltabbo.in/api/lookup?key=Sahil&mobile=${mobile}`;
        
        const response = await fetch(originalUrl, {
            method: 'GET',
            headers: {
                'Content-Type': 'application/json'
            }
        });

        if (!response.ok) {
            return res.status(500).json({
                status: false,
                error: 'Original API returned error',
                code: response.status
            });
        }

        const data = await response.json();

        // Clean the data - remove unwanted fields
        const unwantedFields = ['telegram', 'channel', 'credit', 'api_info'];
        unwantedFields.forEach(field => {
            if (data[field]) delete data[field];
        });

        // Remove unwanted fields from records
        if (data.data && Array.isArray(data.data)) {
            data.data = data.data.map(record => {
                const cleanRecord = { ...record };
                delete cleanRecord.id;
                delete cleanRecord.alt_number;
                return cleanRecord;
            });
        }

        // Add your branding
        data.credit = 'GENIUS HACKER 29 API';
        data.developer = 'ADIBHAI';
        data.youtube = 'https://youtube.com/@geniushacker29';

        // Send success response
        return res.status(200).json(data);

    } catch (error) {
        console.error('API Error:', error);
        
        // Handle specific errors
        if (error.message.includes('fetch')) {
            return res.status(503).json({
                status: false,
                error: 'Connection error - API unreachable'
            });
        }

        if (error.message.includes('timeout')) {
            return res.status(504).json({
                status: false,
                error: 'Request timeout - please try again'
            });
        }

        return res.status(500).json({
            status: false,
            error: 'Internal server error',
            message: error.message
        });
    }
}
